src/Geometry.cpp

```diff
@@ -121,7 +124,14 @@ Geometry::Draw(const Camera& aCamera, const Matrix& aModelTransform) {
     if (kUseTextureCoords) {
       VRB_GL_CHECK(glEnableVertexAttribArray((GLuint)m.renderState->AttributeUV()));
     }
-    VRB_GL_CHECK(glDrawElements(GL_TRIANGLES, m.triangleCount * 3, GL_UNSIGNED_SHORT, 0));
+    const int32_t maxLength = m.triangleCount * 3;
+    if (m.rangeLength == 0) {
+      VRB_GL_CHECK(glDrawElements(GL_TRIANGLES, maxLength, GL_UNSIGNED_SHORT, 0));
+    } else if (m.rangeStart >= maxLength || (m.rangeStart + m.rangeLength) > maxLength) {
+      VRB_WARN("Invalid geometry range (%u-%u). Max geometry length %d", m.rangeStart, m.rangeLength + m.rangeLength, maxLength);
+    } else {
+      VRB_GL_CHECK(glDrawElements(GL_TRIANGLES, m.rangeLength, GL_UNSIGNED_SHORT, (void*)(m.rangeStart * sizeof(GLushort))));
+    }
```

Comment:
I don't want to nit pick (but I will 😅). I think I would prefer it be structure one of these two ways:
```
    if (m.rangeLength == 0) {
      VRB_GL_CHECK(glDrawElements(GL_TRIANGLES, maxLength, GL_UNSIGNED_SHORT, 0));
    } else if ((m.rangeStart + m.rangeLength) <= maxLength) {
      VRB_GL_CHECK(glDrawElements(GL_TRIANGLES, m.rangeLength, GL_UNSIGNED_SHORT, (void*)(m.rangeStart * sizeof(GLushort))));
    } else {
      VRB_WARN("Invalid geometry range (%u-%u). Max geometry length %d", m.rangeStart, m.rangeLength + m.rangeLength, maxLength);
    }
```
or
```
    if ((m.rangeStart + m.rangeLength) <= maxLength) {
      const uint32_t length = m.rangeLength ? m.rangeLength : maxLength;
      VRB_GL_CHECK(glDrawElements(GL_TRIANGLES, length, GL_UNSIGNED_SHORT, (void*)(m.rangeStart * sizeof(GLushort))));
    } else {
      VRB_WARN("Invalid geometry range (%u-%u). Max geometry length %d", m.rangeStart, m.rangeLength + m.rangeLength, maxLength);
    }
```

Reply:
np, PR updated!